added the color brewer qualitative palettes

stvn66 opened the pull request:
fixes #4758 
- [ added the qualitative palettes from ColorBrewer ] issues: 
- [ updated and modified the palette test (now checking the last int from the string name instead of the first int ] tests added / passed
- [ ] release document entry (if new feature or API change)

canavandl commented:
Two line length errors (https://travis-ci.org/bokeh/bokeh/jobs/145278743#L1404). So close!

canavandl commented on examples/charts/file/palettes.py:
```diff
@@ -26,7 +27,11 @@
-                                 ("Viridis9", Viridis9)])
+                                 ("Viridis9", Viridis9), ("Accent8", Accent8),
+                                 ("Dark8", Dark8), ("Paired9", Paired9),
+                                 ("Paired9", Paired9), ("Pastel1_9", Pastel1_9),
```
Paired9 was added twice.

canavandl commented on examples/charts/file/palettes.py:
```diff
@@ -26,7 +27,11 @@
                                  ("Magma9", Magma9), ("Plasma9", Plasma9),
-                                 ("Viridis9", Viridis9)])
+                                 ("Viridis9", Viridis9), ("Accent8", Accent8),
+                                 ("Dark8", Dark8), ("Paired9", Paired9),
```
I think Dark is actually called Dark2 (though there doesn't seem to exist a Dark1): https://jiffyclub.github.io/palettable/colorbrewer/qualitative/#dark2_3

canavandl commented:
Two tiny comments, otherwise LGTM (the failing integration tests aren't your fault, but because the SauceLab uploads don't work for non-Bokeh repo owners).

Thanks so much for your contribution. It was nice meeting you at the sprints today.

caryan commented:
@stvn66 thanks for sending the PR.   The only other thing that came up in #4758 is that we're missing a license/attribution for the Brewer colours.  All I can find from Cynthia's  [webpage](http://www.personal.psu.edu/cab38/ColorBrewer/ColorBrewer_updates.html) is that it is Apache v2 licensed and we should add attribution.

I had this in my local copy:

```python
# This product includes color specifications and designs developed by Cynthia Brewer
# (http://colorbrewer2.org/).  The Brewer colormaps are licensed under the Apache v2 license.
# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0 
```

stvn66 commented:
I just replaced Dark with Dark2, added the license for the Brewer colormaps to `palettes.py`, and removed the duplicate `Paired9` in the example file.

@canavandl, It was a pleasure to get to work on the project and become more familiar with the code base.

canavandl commented on bokehjs/src/coffee/palettes/palettes.coffee:
```diff
@@ -454,8 +454,91 @@ palettes = {
+    Accent8     : [0x7fc97f, 0xbeaed4, 0xfdc086, 0xffff99, 0x386cb0, 0xf0027f, 0xbf5b17, 0x666666]
+  }
+
+  Dark: {
```
These also need to be renamed Dark2, if you have a sec. Otherwise it's ready to merge

bryevdv replied:
friendly ping @stvn66 will you be able to make this last minor change? I would be nice to include this in `0.12.1` if possible, but that's in the next few days.

canavandl commented:
LGTM. thanks again @stvn66
